src/app/services/socket/socket.octoprint.service.ts

```diff
@@ -116,6 +116,27 @@ export class OctoPrintSocketService implements SocketService {
     this.socket.next(payload);
   }
 
+  private handlePluginMessage(pluginMessage: OctoprintPluginMessage) {
+    const plugins = [
+      {
+        check: (plugin: string) => plugin === 'DisplayLayerProgress-websocket-payload'
+          && this.configService.isDisplayLayerProgressEnabled(),
+        handler: (data: unknown) => {
+          this.extractFanSpeed(data as DisplayLayerProgressData);
+          this.extractLayerHeight(data as DisplayLayerProgressData);
+        },
+      },
+      {
+        check: (plugin: string) => ['action_command_prompt', 'action_command_notification'].includes(plugin),
+        handler: (data: unknown) => this.eventSubject.next(data as PrinterNotification),
+      },
+    ];
+
+    plugins.forEach(plugin =>
+      plugin.check(pluginMessage.plugin.plugin) && plugin.handler(pluginMessage.plugin.data)
```

Comment:
That definitely looks complicated (which isn't bad). Just curious: Will call this both check methods from the plugins array? Seems like a nice way to do this, since it's easily extendable 👍

Reply:
Ok I think I understood it. Correct me if I'm wrong:

This will go through top to bottom and call the check method with the plugin string. If the check message returns true it will call the handler in the same object, thus handle the message. Otherwise it will just fall-through until it either matches or reaches the end of the array. Really elegant way of handling this 👍